schemaPool.go

```diff
@@ -168,6 +194,9 @@ func (p *schemaPool) GetDocument(reference gojsonreference.JsonReference) (*sche
 	refToURL.GetUrl().Fragment = ""
 
 	if cachedSpd, ok := p.schemaPoolDocuments[refToURL.String()]; ok {
+		if bsonD, isBsonD := cachedSpd.Document.(bson.D); isBsonD {
```

Comment:
This fixes an issue where the tests were failing here: https://github.com/10gen/gojsonschema/blob/9b64c19c895bb7d38d2526ab22303b2bc8510f46/jsonschema_test.go#L1174